src/engine.py

```
@@ -1308,42 +1313,62 @@ async def _outbound_choose_endpoint(self, dial_phone: str) -> str:
+                resp = await self.ari_client.send_command(
+                    "GET",
+                    f"endpoints/{tech}/{phone}",
+                    tolerate_statuses=[404],
+                )
+                if isinstance(resp, dict) and int(resp.get("status") or 0) == 404:
+                    continue
+                if isinstance(resp, dict) and ("resource" in resp or "technology" in resp or "state" in resp):
+                    self._outbound_pjsip_endpoint_cache[phone] = {"exists": True, "tech": tech, "ts": now}
+                    return f"{tech}/{phone}"
+            except Exception:
+                continue
```

Ruff (0.15.0)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)